### PR TITLE
[WIP] Add JSDoc to evt.common

### PR DESCRIPTION
**Pull request by Bouh**
Just to be sure if everything is correct for now in this first commit.
I'll write the rest of the file in this PR if all is ok.

### CODE REVIEW

File: GDJS/Runtime/events-tools/commontools.js [hunk 1]
```diff
@@ -106,16 +106,38 @@ gdjs.evtTools.common.logicalNegation = function(bool) {
...
+ * @param {number} max Maximum value.
+ * @returns {number} The new value.
+ */
 gdjs.evtTools.common.clamp = function(x, min, max) {
   return Math.min(Math.max(x, min), max);
 };
 
+/**
+ * Hyperbolic arccosine
+ * @memberof gdjs.evtTools
```
Comment (arthuro555) [inline GDJS/Runtime/events-tools/commontools.js:122]: You don't need that memberof tag. Plus it is inconsistent (was not added above

File: GDJS/Runtime/events-tools/commontools.js [hunk 2]
```diff
@@ -106,16 +106,38 @@ gdjs.evtTools.common.logicalNegation = function(bool) {
...
+ * @class common
```
Comment (arthuro555) [inline GDJS/Runtime/events-tools/commontools.js:123]: The class tag is used to say a class is defined here, not that it is a member. Also, common is a namespace not a class.

File: GDJS/Runtime/events-tools/commontools.js [hunk 3]
```diff
@@ -106,16 +106,38 @@ gdjs.evtTools.common.logicalNegation = function(bool) {
   return !bool;
 };
 
+/**
+ * Limit an value in a range
+ * @param {number} x Value.
+ * @param {number} min Minimum value.
+ * @param {number} max Maximum value.
+ * @returns {number} The new value.
```
Comment (arthuro555) [inline GDJS/Runtime/events-tools/commontools.js:114]: This is also a little inconsistency: you included a "the" where you didn't for the param tags

**Comment by Bouh** [comment]
It's fixed, great thank you!
I can write the rest of this file now :)

**Comment by arthuro555** [comment]
Be careful, class and namespace tags are not used to tell it's a member but to say it's the class *definition*. You should remove them 😉

**Comment by 4ian** [comment]
Thanks both! What I see so far seems good (namespace is better than class here, we don't have a class here. memberOf is also not useful indeed).
I see that the auto-formatting did add some space before "function". I think it's the Visual Studio Code builtin autoformattting - or maybe a different version of prettier? In any case when I launch the formatting on this file on my computer I don't have these extra spaces. Could you double check that? :)

**Comment by Bouh** [comment]
I use lastest build of VScode, prettier and eslint. i've disabled eslint.
My default formater is `esbenp.prettier-vscode`
Nothing about spacing before function in parameters, and all peoples want add this damn space. Like if this is now common for everyone.

**Comment by 4ian** [comment]
npm run format is only for the newIDE, GDJS does not have it... but I wonder if we should add it (next to `npm run check-types`).

**Comment by 4ian** [comment]
Seems that the new Prettier extension  that you used adds tabs instead of 2 spaces 😬

**Comment by Bouh** [comment]
We should add it somewhere in the Code of conduct or in an readme few words about 2 spaces instead tabs.

**Comment by 4ian** [comment]
> We should add it somewhere in the Code of conduct or in an readme few words about 2 spaces instead tabs.

Ideally all the code styling is automated so we don't even need to mention it beside running the auto formatting.

> And a list of requirement.
> Prettier and which extension, flow, grunt, npm, yarn ...

There are recommended tools here:
https://github.com/4ian/GDevelop/blob/master/newIDE/README.md#recommended-tools-for-development

but we can expand it if needed.

But I think here with an autoformatting command for GDJS (and extensions), all of our problems would be solved.

File: GDJS/Runtime/events-tools/commontools.js [hunk 4]
```diff
@@ -31,24 +30,15 @@ gdjs.evtTools.common.getVariableString = function(variable) {
...
-gdjs.evtTools.common.sceneVariableExists = function(
-  runtimeScene,
-  variableName
-) {
+gdjs.evtTools.common.sceneVariableExists = function(runtimeScene, variableName) {
   return runtimeScene.getVariables().has(variableName);
 };
 
 /**
  * @private
```
Comment (arthuro555) [inline GDJS/Runtime/events-tools/commontools.js:38]: Maybe try adding a description to the private fields explaining their purpose and why they are private could be useful (for devs).

**Comment by 4ian** [comment]
Made a few fixes, merging now! Thanks :)

**Comment by Bouh** [comment]
> memberOf is also not useful indeed).

Maybe this is useful, because unless the GDJS docs isn't refresh, i can't see [gdjs.evtTools.common](https://docs.gdevelop-app.com/GDJS%20Runtime%20Documentation/gdjs.evtTools.html)

I can see network, and it's define like this:

```
/**
 * @memberof gdjs.evtTools
 * @namespace network
 */
gdjs.evtTools.network = gdjs.evtTools.network || {};
```

While common is:

```
/**
 * @namespace common
 * @static
 * @private
 */
gdjs.evtTools.common = gdjs.evtTools.common || {};
```

**Comment by arthuro555** [comment]
The problem here would be more the private tag than the memberof I think

**Comment by 4ian** [comment]
It's indeed expected that members marked as private are not shown in the documentation.
You can try to regenerate the doc yourself by running `npm run generate-doc` in GDJS folder :)

**Comment by Bouh** [comment]
I tried to generate the documentation after reading the [readme here.](https://github.com/4ian/GDevelop/tree/master/GDJS/docs)
I expect to be with my console in "GDevelop\GDJS\docs"
And i run the commands.

But nothing happend, i've asked to arthuro555 why, and i need to be at the root "GDevelop"
and execute the commands.
No very clear for me. I was fooled by the folder GDJS in GDJS, so for me the GDJS on the readme was "GDevelop\GDJS\GDJS".

**Comment by 4ian** [comment]
I clarified https://github.com/4ian/GDevelop/blob/master/GDJS/docs/Readme.md

**Pull request merged**